include/rapidcheck/gen/Text.hpp

```diff
@@ -64,11 +95,52 @@ Gen<T> character() {
   };
 }
 
+
+template <typename T>
+Gen<T> unicodeCodepoint() {
+	return [](const Random &random, int size) {
+		auto stream = ::rc::detail::bitStreamOf(random);
+
+		return shrinkable::shrinkRecur(rc::detail::generateCodePoint<T>(stream),
+			&shrink::unicodeCodepoint<T>);
+	};
+}
+
+template <typename Container>
+Gen<Container> unicodeCodepoints()
```

Comment:
In the spirit of non-templatization, I think you could go for `std::vector<std::uint32_t>>´ all the way down here. That way, we can spare users the compile time cost of template instantiation.